File: Module5/2.Post/FSharpWeb/Controllers/CustomersController.fs
namespace FSharpWeb.Controllers

open System.Web.Mvc
open Services

type CustomersController() =
    inherit Controller()

    let service = CustomerService()
    
    member x.Index () = 
        x.View(service.GetCustomers ())

    member x.Post (ids: int seq) = 
        service.UpgradeCustomers(ids);
        x.RedirectToAction("Index")

File: Module5/2.Post/CSharpWeb/Controllers/CustomersController.cs
using System.Linq;
using System.Web.Mvc;
using CSharpWeb.Models;
using Services;

namespace CSharpWeb.Controllers
{
    public class CustomersController : Controller
    {
        private readonly CustomerService service;

        public CustomersController()
        {
            service = new CustomerService();
        }

        public ActionResult Index()
        {
            var customers = service.GetCustomers();
            var model = customers.Select(c => new CustomerViewModel
            {
                Id = c.Id,
                Credit = c.Credit,
                IsVip = c.IsVip
            });
            return View(model);
        }

        [HttpPost]
        public ActionResult Post(int[] ids)
        {
            service.UpgradeCustomers(ids);
            return RedirectToAction("Index");
        }
    }
}